src/ProjectTemplates/test/BlazorTemplateTest.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+using Templates.Test.Helpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Templates.Test
+{
+    public abstract class BlazorTemplateTest : LoggedTest
+    {
+        public BlazorTemplateTest(ProjectFactoryFixture projectFactory)
+        {
+            ProjectFactory = projectFactory;
+        }
+
+        public ProjectFactoryFixture ProjectFactory { get; set; }
+
+        private ITestOutputHelper _output;
+        public ITestOutputHelper Output
+        {
+            get
+            {
+                if (_output == null)
+                {
+                    _output = new TestOutputLogger(Logger);
+                }
+                return _output;
+            }
+        }
+
+        public abstract string ProjectType { get; }
+
+        protected async Task<Project> CreateBuildPublishAsync(string projectName, string auth = null, string[] args = null, string targetFramework = null, bool serverProject = false, bool onlyCreate = false)
+        {
+            // Additional arguments are needed. See: https://github.com/dotnet/aspnetcore/issues/24278
+            Environment.SetEnvironmentVariable("EnableDefaultScopedCssItems", "true");
+
+            var project = await ProjectFactory.GetOrCreateProject(projectName, Output);
+            if (targetFramework != null)
+            {
+                project.TargetFramework = targetFramework;
+            }
+
+            var createResult = await project.RunDotNetNewAsync(ProjectType, auth: auth, args: args);
+            Assert.True(0 == createResult.ExitCode, ErrorMessages.GetFailedProcessMessage("create/restore", project, createResult));
+
+            if (!onlyCreate)
+            {
+                var targetProject = project;
+                if (serverProject)
+                {
+                    targetProject = GetSubProject(project, "Server", $"{project.ProjectName}.Server");
+                }
+
+                var publishResult = await targetProject.RunDotNetPublishAsync(noRestore: false);
+                Assert.True(0 == publishResult.ExitCode, ErrorMessages.GetFailedProcessMessage("publish", targetProject, publishResult));
+
+                // Run dotnet build after publish. The reason is that one uses Config = Debug and the other uses Config = Release
+                // The output from publish will go into bin/Release/netcoreappX.Y/publish and won't be affected by calling build
+                // later, while the opposite is not true.
```

Comment:
This is patently false btw.

Reply:
Remove or is there updated text?

Reply:
Remove, you should be able to build and publish in any Configuration.

Reply:
Cool, I deleted it from the current blazor template tests in quarantine as well, so we no longer build after publish